Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.mm

```diff
@@ -127,7 +179,54 @@ Ref<Texture> DeviceImpl::createTexture(const TextureDescriptor& descriptor)
         descriptor.sampleCount,
     };
 
-    return TextureImpl::create(wgpuDeviceCreateTexture(backing(), &backingDescriptor), descriptor.format, descriptor.dimension, m_convertToBackingContext);
+    IOSurfaceRef ioSurface = surface ? static_cast<const SurfaceImpl*>(surface)->surface() : nullptr;
```

Comment:
I would understand that the function createTexture(descr) is just a normal texture creation function.

I'd imagine you'd need a new function for "create texture that is backed by the surface". Perhaps conceptually the same as the one you add in the idl, gpusurface.createTexture() or something?